packages/ui-components/src/Tooltip/Tooltip.tsx

```diff
@@ -67,36 +70,48 @@ export const Tooltip: FunctionComponent<TooltipProps> = ({
       ) : (
         (child as ReactElement)
       ),
-      {
-        ref: setReferenceElement,
-        onMouseOver: () => {
-          popperElement.setAttribute("data-show", "");
-        },
-        onMouseOut: () => {
-          if (visible) return;
-          popperElement.removeAttribute("data-show");
-        },
-      },
     );
   });
 
+  const timers: any = [];
   return (
-    <>
+    <div
+      ref={setReferenceElement}
+      onMouseOver={() => {
+        const t = setTimeout(() => {
+          popperElement.setAttribute("data-show", "");
+        }, timeout);
+        timers.push(t);
+      }}
+      onMouseLeave={() => {
+        timers.forEach((t: any) => {
```

Comment:
and you can replace this any with

```suggestion
       timers.forEach((t: ReturnType<typeof setTimeout>) => {
```